runtime/onert/core/include/ir/Operation.h

```diff
@@ -28,7 +28,7 @@ namespace onert
 namespace ir
 {
 
-class Operation : public IOperation
+class Operation : virtual public IOperation
```

Comment:
@ragmani Could you please add comment why virtual inheritane is introduced? As I remember, `Loss` inherits `ITrainableOperation` and `Operation` (I am not sure.).

Reply:
You are right. It is a c++ technique that can be used to avoid the [diamond problem](https://en.wikipedia.org/wiki/Diamond_problem) by clarifying ambiguity over which ancestor class to use.
I'll add this comment.

Reply:
@ragmani I know `virtual` keyword is required to solve diamond problem. I ask you to leave why diamond problem is introduced. Maybe

```
// Diamond problem is introduced because Loss inherits ITrainableOperation and Operation.
```